### PR TITLE
Android: Support to enumerate and select Camera devices

### PR DESCRIPTION
**Pull request by cristianoccazinsp**
# Summary
This update attempts to bring a very useful feature to support selecting from multiple back/front cameras (e.g., for phones with wide lenses cameras, or multi cameras). 

The update implements a ```getCameraIdsAsync``` methods to enumerate all available cameras, and adds a ```cameraId``` property to the RNCamera component to set it.

This should be 100% backwards compatible. A null cameraId value (the default), will work as it currently does by selecting the camera based on the type value. However, setting an ID will override the type value and select that camera.

I've implemented the change for both Camera1 and Camera2 APIs, but I could really use some extra pair of eyes to test and review the changes since I'm not 100% confident with the internals of the library and I might have missed some stuff.

Lastly, I'm unsure of the flash behaviour if the front facing camera is used. I believe it won't be possible to use a "front" flash right now.

For now, these changes will be Android only. It makes sense considering that only iPhone 11 will support multiple cameras. I also lack Objective-C knowledge and camera internals like to do this change.

Related issue: https://github.com/react-native-community/react-native-camera/issues/2379

## Test Plan
Tested on Pixel 2 (android 9 and 10), Moto G5 (Android 8), LG V20

### What's required for testing (prerequisites)?
Real device

### What are the steps to reproduce (after prerequisites)?
-

## Compatibility

| OS      | Implemented |
| ------- | :---------: |
| iOS     |    ❌     |
| Android |    ✅    |

## Checklist

- [X] I have tested this on a device and a simulator
- [X] I added the documentation in `README.md`
- [X] I mentioned this change in `CHANGELOG.md`
- [X] I updated the typed files (TS and Flow)
- [X] I added a sample use of the API in the example project (`example/App.js`)

### CODE REVIEW

**Comment by cristianoccazinsp** [comment]
Pushed an additional update to remove the requirement from the camera to be running/active. We can still query for this info as long as it is instantiated.

File: android/src/main/java/com/google/android/cameraview/Camera1.java [hunk 1]
```diff
@@ -693,14 +739,25 @@ public Size getPreviewSize() {
...
-                return;
+        if(_mCameraId == null){
+            for (int i = 0, count = Camera.getNumberOfCameras(); i < count; i++) {
+                Camera.getCameraInfo(i, mCameraInfo);
+                if (mCameraInfo.facing == mFacing) {
+                    mCameraId = i;
+                    return;
+                }
+            }
+            mCameraId = 0; // default to the only available camera
```
Comment (cristianoccazinsp) [inline android/src/main/java/com/google/android/cameraview/Camera1.java]: @sibelius  Can you comment on these lines? I wonder if we need to do something different here actually to handle the case where a camera is not found.  I think I overlooked this one and it should have been INVALID_CAMERA_ID instead. Thoughts?
Need to test on a device with no cameras, or just 1 camera.

Comment (sibelius) [reply android/src/main/java/com/google/android/cameraview/Camera1.java]: hard to find a device without cameras

what happen when you throw? can an user get this error using an ErrorBoundary?

`0` is always the first camera?

won't break the current behavior?

Comment (cristianoccazinsp) [reply android/src/main/java/com/google/android/cameraview/Camera1.java]: We can/should probably mimic what Camera2 does and raise an error if there are 0 cameras, but set the camera to the first one available if we didn't find one: https://github.com/react-native-community/react-native-camera/blob/master/android/src/main/java/com/google/android/cameraview/Camera2.java#L688

How does that sound? I believe the Camera1 code originally would just set the camera ID to be the invalid one and you would get no image at all in this case (if lets say you don't have a back camera). It's a very specific case.

I was thinking about something like this:

```
int count = Camera.getNumberOfCameras();
            if(count == 0){
                throw new RuntimeException("No camera available.");
            }

            for (int i = 0; i < count; i++) {
                Camera.getCameraInfo(i, mCameraInfo);
                if (mCameraInfo.facing == mFacing) {
                    mCameraId = i;
                    return;
                }
            }
            // no camera found, set the one we have
            mCameraId = 0;
            Camera.getCameraInfo(mCameraId, mCameraInfo);
```

Comment (sibelius) [reply android/src/main/java/com/google/android/cameraview/Camera1.java]: this part is weird

// no camera found, set the one we have
mCameraId = 0;

but it makes sense, you are looking for the camera facing the correct facing the user selected

Comment (cristianoccazinsp) [reply android/src/main/java/com/google/android/cameraview/Camera1.java]: The original code would simply set the camera to be the INVALID_ID one, but that behaviour is different from the Camera2 API which simply sets the first available camera if the code wasn't able to find the back/front facing camera. Do you think it makes sense to make both APIs consistent?

Comment (sibelius) [reply android/src/main/java/com/google/android/cameraview/Camera1.java]: sure

Comment (cristianoccazinsp) [reply android/src/main/java/com/google/android/cameraview/Camera1.java]: I've just added a commit for that.

**Comment by sibelius** [comment]
tests are failing

**Comment by cristianoccazinsp** [comment]
My bad, missed this when moving the file from my test folder to the repo. Should be good now.

**Comment by cristianoccazinsp** [comment]
Doing more tests to the Camera2 API, please do not merge this yet since I've found a few more issues (some not even related to my changes, but will fix them since I'm at it)

**Comment by cristianoccazinsp** [comment]
Did a few more fixes to Camera2.

**Comment by sibelius** [comment]
@cristianoccazinsp is this good enough now?

**Comment by cristianoccazinsp** [comment]
Haven't had issues or crashes so far, so I would say it is good enough. It
also seem to work fine in devices with wide angle lenses (LG V20 and V30)
for example.

**Pull request merged**